auto_nag/scripts/inactive_ni_pending.py

```diff
@@ -0,0 +1,289 @@
+                )
+            ]
+
+        res = {}
+        skiplist = self.get_auto_ni_skiplist()
+        for bugid, bug in bugs.items():
+            if (
+                bugid not in inactive_requestee_bugs
+                or bug["triage_owner"] in inactive_users
+                or bug["triage_owner"] in skiplist
+            ):
+                continue
+
+            inactive_ni = get_inactive_ni(bug)
+            if len(inactive_ni) == 0:
+                continue
+
+            bug = {
+                **bug,
+                "inactive_ni": inactive_ni,
+                "inactive_ni_count": len(inactive_ni),
+                "action": self.get_action_type(bug, inactive_ni),
+            }
+            res[bugid] = bug
+            self.add_action(bug)
+
+        return res
+
+    @staticmethod
+    def get_action_type(bug, inactive_ni):
+        """
+        Determine if should forward needinfos to the triage owner, clear the
+        needinfos, or close the bug.
+        """
+
+        if (
+            bug["priority"] in HIGH_PRIORITY
+            or bug["severity"] in HIGH_SEVERITY
+            or bug["creation_time"] >= RECENT_BUG_LIMIT
+        ):
+            return NeedinfoAction.FORWARD
+
+        if bug["severity"] == "--":
+            if (
+                len(bug["needinfo_flags"]) == 1
+                and inactive_ni[0]["requestee"] == bug["creator"]
+                and not inactive_ni[0]["canconfirm"]
+            ):
+                return NeedinfoAction.CLOSE_BUG
+
+            return NeedinfoAction.FORWARD
+
+        return NeedinfoAction.CLEAR
+
+    @staticmethod
+    def _clear_inactive_ni_flags(bug):
+        return [
+            {
+                "id": flag["id"],
+                "status": "X",
+            }
+            for flag in bug["inactive_ni"]
+        ]
+
+    @staticmethod
+    def _needinfo_triage_owner_flag(bug):
+        return [
+            {
+                "name": "needinfo",
+                "requestee": bug["triage_owner"],
+                "status": "?",
+                "new": "true",
+            }
+        ]
+
+    def add_action(self, bug):
+        users_num = len(set([flag["requestee"] for flag in bug["inactive_ni"]]))
+
+        if bug["action"] == NeedinfoAction.FORWARD:
+            request_from_triage_owner = (
+                "could you please set the severity or close the bug?"
+                if bug["severity"] == "--"
+                else "could you have a look please?"
+            )
+            autofix = {
+                "flags": (
+                    self._clear_inactive_ni_flags(bug)
+                    + self._needinfo_triage_owner_flag(bug)
+                ),
+                "comment": {
+                    "body": (
+                        f'Redirect { plural("a needinfo that is", bug["inactive_ni"], "needinfos that are") } pending on { plural("an inactive user", users_num, "inactive users") } to the triage owner.'
```

Comment:
Can we add the reason why we forward? This will reduce complaints (e.g. if we say things like "forwarding because the bug has high severity", "forwarding because the bug was recently touched", etc.).

Reply:
Done in 059ffe4f539cc6b1b06ba610dc3b34426243447f